python-backend/app/api/search/search/resources/search.py

```diff
@@ -0,0 +1,132 @@
+        search_types = search_types.split(',') if search_types else search_targets.keys()
+
+        reg_exp = regex.compile(r'\'.*?\' | ".*?" | \S+ ', regex.VERBOSE)
+        search_terms = reg_exp.findall(search_term)
+        search_terms = [term.replace('"', '') for term in search_terms]
+
+        with ThreadPoolExecutor(max_workers=50) as executor:
+            task_list = []
+            for type, type_config in search_targets.items():
+                if type in search_types:
+                    task_list.append(
+                        executor.submit(execute_search, app, search_results, search_term,
+                                        search_terms, type, type_config))
+            for task in as_completed(task_list):
+                try:
+                    data = task.result()
+                except Exception as exc:
+                    current_app.logger.error(
+                        f'generated an exception: {exc} with search term - {search_term}')
+
+        grouped_results = {}
+        for result in search_results:
+            if (result.result['id'] in grouped_results):
+                grouped_results[result.result['id']].score += result.score
+            else:
+                grouped_results[result.result['id']] = result
+
+        top_search_results = list(grouped_results.values())
+        top_search_results.sort(key=lambda x: x.score, reverse=True)
+
+        all_search_results = {}
+
+        for type in search_types:
+            top_search_results_by_type = {}
+
+            max_results = 5
+            if len(search_types) == 1:
+                max_results = 50
+
+            for result in top_search_results:
+                if len(top_search_results_by_type) > max_results:
+                    break
+                if result.type == type:
+                    top_search_results_by_type[result.result['id']] = result
+
+            full_results = db.session.query(search_targets[type]['model']).filter(
+                search_targets[type]['primary_column'].in_(
+                    top_search_results_by_type.keys())).all()
+
+            for full_result in full_results:
+                top_search_results_by_type[getattr(
+                    full_result, search_targets[type]['id_field'])].result = full_result
+
+            all_search_results[type] = list(top_search_results_by_type.values())
+
+        return {'search_terms': search_terms, 'search_results': all_search_results}
+
+
+def append_result(search_results, search_term, type, item, id_field, value_field, score_multiplier):
+
+    if getattr(item, value_field).lower().startswith(search_term.lower()):
+        score_multiplier = score_multiplier * 3
+
+    if getattr(item, value_field).lower() == search_term.lower():
+        score_multiplier = score_multiplier * 10
+
+    search_results.append(
+        SearchResult(
+            getattr(item, 'score') * score_multiplier, type, {
```

Comment:
i don't believe there is any reason to use getattr unless you have defined a default value (3rd parameter), i think these should be item.value_field.